diff --git a/node_modules/metro-react-native-babel-transformer/src/index.js b/node_modules/metro-react-native-babel-transformer/src/index.js
index 3b7bdb7..f969e4c 100644
--- a/node_modules/metro-react-native-babel-transformer/src/index.js
+++ b/node_modules/metro-react-native-babel-transformer/src/index.js
@@ -128,7 +128,19 @@ function buildBabelConfig(filename, options, plugins = []) {
   };
   let config = { ...babelRC, ...extraConfig }; // Add extra plugins
 
-  const extraPlugins = [];
+  const extraPlugins = [
+    [
+      "search-and-replace",
+      {
+        "rules": [
+          {
+            "search": "import electron from 'electron'",
+            "replace": "const electron = null"
+          }
+        ]
+      }
+    ]
+  ];
 
   if (options.inlineRequires) {
     extraPlugins.push(inlineRequiresPlugin);
